python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -127,14 +127,21 @@ export default function EditorContainer({
         async (
           promptName: string,
           newPrompt: Prompt,
-          onSuccess?: (aiconfigRes: AIConfig) => void
+          onSuccess: (aiconfigRes: AIConfig) => void,
+          onError: (err: unknown) => void
         ) => {
-          const serverConfigRes = await updatePromptCallback(
-            promptName,
-            newPrompt
-          );
-          if (serverConfigRes && onSuccess) {
-            onSuccess(serverConfigRes.aiconfig);
+          try {
+            const serverConfigRes = await updatePromptCallback(
+              promptName,
+              newPrompt
+            );
+            if (serverConfigRes && onSuccess) {
+              onSuccess(serverConfigRes.aiconfig);
```

Comment:
Checking if this should be `serverConfigRes`? Or if `.aiconfig` indexing here is valid

Reply:
aiconfig indexing is valid for our server responses. I can add the extra check to be sure